src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -3,47 +3,75 @@ import { default as styled } from 'styled-components';
 import { InternalTokensProp } from '../../theme';
 import { withSuomifiDefaultProps } from '../../theme/utils';
 import { baseStyles } from './SearchInput.baseStyles';
-import { TextInput, TextInputProps } from '../TextInput/TextInput';
+import { baseStyles as inputBaseStyles } from '../TextInput/TextInput.baseStyles';
+import { BaseTextInput } from '../../../components/Form/TextInput';
+import { TextInputProps, textInputClassNames } from '../TextInput/TextInput';
 import { Icon } from '../../Icon/Icon';
 import classnames from 'classnames';
 import { Omit } from '../../../utils/typescript';
+import { logger } from '../../../utils/logger';
 
 const baseClassName = 'fi-search-input';
 const inputContainerBaseClassName = `${baseClassName}_input-container`;
 const inputBaseClassName = `${baseClassName}_input`;
 const iconBaseClassName = `${baseClassName}_icon`;
 
-export interface SearchInputProps extends Omit<TextInputProps, 'variant'> {}
+export interface SearchInputProps extends Omit<TextInputProps, 'variant'> {
+  /** Placeholder text for search field. Must match labelText */
+  placeholder?: string;
```

Comment:
I think previously it was so that if label was made visible the placeholder disappears. At the moment both can be visible at the same time if desired, but the placeholder gets its value from labelText in any case. As mentioned in another comment, a boolean might be a clearer way to implement this. Or we can have it as it was, with just the options for a hidden label with labelText as placeholder or a visible label without a placeholder.

Reply:
Yeah I would suggest not allowing to set `placeholder`-prop and HTML placeholder-attribute is always from labelText if conditions met.